xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/less/buttons.less

```
@@ -9,72 +9,79 @@
   #gradient > .vertical-three-colors(lighten(@color, 10%), @color, 50%, darken(@color, 5%));
 }
 
+.btn-background(@color) {
+  background-color: @color;
+}
+
 .btn-border(@color) {
   border: 1px solid @color;
   border-bottom-color: darken(@color, 10%);
-  .box-shadow(~"inset 0 1px 0 rgba(255,255,255,.2), 0 1px 2px rgba(0,0,0,.05)");
+}
+
+.btn-no-border(@color) {
+  border: 1px solid @color;
```

```suggestion
  border: 0;
```
Is there a point to keeping this solid border even in the no-border mixin? From what I understood, it's always the same color as the background, but I don't think we should need a color argument to define a no-border element

---

I should indeed document that, but without having a border the same color as the background, the buttons without border are 2px smaller than the ones with a border.
I felt like the easiest way was to have a border the same color as the background.

---

I don't know this property too well, but maybe you should consider using a different `box-sizing` value to solve this inconsistency :)

https://developer.mozilla.org/en-US/docs/Web/CSS/box-sizing